Add auto complete for affinity topology keys

bashofmann opened the pull request:
When configuring a pod affinity on a workload, you have to enter a topologyKey. This can be any node label key. For more details see https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity.

To make the configuration easier, this PR changes the input field to a select which offers all existing node label keys, with the option to also add keys that are not present as labels yet.

![Bildschirmfoto 2022-03-02 um 16 22 18](https://user-images.githubusercontent.com/243056/156392738-683e6357-1265-450a-ac20-d66d8fefda1c.png)

richard-cox commented on utils/array.js:
```diff
@@ -153,3 +153,7 @@ export function uniq(ary) {
   return out;
 }
+
+export function getUniqueLabelKeys(aryResources) {
```
This could be defined a little clearer

```
  return aryResources
    .map(res => Object.keys(res.metadata.labels))
    .flat()
    .filter((value, index, self) => self.indexOf(value) === index)
    .sort();
```

or to reduce some of the iterations it could be 

```
  const uniqueObj = aryResources.reduce((res, r) => {
    Object.keys(r.metadata.labels).forEach(l => (res[l] = true));

    return res;
  }, {});

  return Object.keys(uniqueObj).sort();
```

or you could do something with a `Set` but by the time that's converted to an array to sort i'm not sure it's worth it.

richard-cox commented on components/form/LabeledSelect.vue:
```diff
@@ -55,6 +55,10 @@ export default {
       default: null,
       type:    String
     },
+    pushTags: {
```
I couldn't see this used in the PodAffinity LabeledSelect. Was it to fix `./chart/monitoring/StorageClassSelector.vue`?

bashofmann replied:
This accidentally got in. It doesn't do any harm, but since it's not necessary for this PR, I'll remove it.

richard-cox reviewed (changes requested):
Two minor comments, otherwise looking good.

I did notice that once a value is provided it can't be removed, the user would need to remove the entire affinity. However as the field is required I don't think that's an issue.

bashofmann commented:
I addressed both comments and also added a test for the utils function.

> I did notice that once a value is provided it can't be removed, the user would need to remove the entire affinity. However as the field is required I don't think that's an issue.

Agreed, if you wanted to remove it, you would remove the whole selector, otherwise you'd just change the value.
